jobs/atc/spec

```diff
@@ -559,3 +559,15 @@ properties:
     description: |
       AWS SecretsManager secret name template used to resolve team specific secrets.
     default: /concourse/{{.Team}}/{{.Secret}}
+
+  build_log_retention.default:
+    description: |
+      Default (can be overriden by job) number of build logs to retain,
+      0 means retain all (database will grow indefinitely).
+    default: 0
```

Comment:
Rather than having empty defaults at the BOSH spec level, could we omit the defaults here and just conditionally pass the flags? The 0-value interpretation is kind of an implementation detail and could be confusing once it's exposed at higher levels, since really 0 means "don't do this feature".

iirc we tend to just conditionally pass flags for all the other properties.

Reply:
Thanks @vito - fix as suggested (I think).